tests/softreboot_test.go

```diff
@@ -64,114 +73,106 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 	})
 
 	Context("Soft reboot VMI", func() {
-
 		var vmi *v1.VirtualMachineInstance
 
-		runVMI := func(withGuestAgent bool, ACPIEnabled bool) {
-			if withGuestAgent {
-				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
-			} else {
-				vmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskCirros))
-				tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
-			}
-			if !ACPIEnabled {
-				vmi.Spec.Domain.Features = &v1.Features{
-					ACPI: v1.FeatureState{Enabled: &ACPIEnabled},
-				}
-			}
-			tests.RunVMIAndExpectLaunch(vmi, 360)
-		}
-
 		When("soft reboot vmi with agent connected via API", func() {
 			It("should succeed", func() {
-				runVMI(true, false)
+				vmi = libvmi.NewTestToolingFedora(withoutACPI())
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 360)
```

Comment:
nit: you could inline these two. 
```golang
vmi = tests.RunVMIAndExpectLaunch(
    libvmi.NewTestToolingFedora(withoutACPI()), 
    360)
```

You're welcome to preserve your alternative.